src/json.cxx

```diff
@@ -15,3 +16,103 @@ std::string json_to_string(rapidjson::Document const &jd) {
   jd.Accept(w);
   return b1.GetString();
 }
+
+/// Merge the json structure v2 into a copy of v1.
+/// If the new key in v2 is not a object, it replaces the key in v1.
+/// Elements of arrays are also not merged, but the new array fully replaces
+/// the old one.
+rapidjson::Document merge(rapidjson::Value const &v1,
+                          rapidjson::Value const &v2) {
+  using namespace rapidjson;
+  using std::string;
+  using std::deque;
+  Document ret;
+  auto &a = ret.GetAllocator();
+  ret.CopyFrom(v1, a);
+  if (!v1.IsObject() || !v2.IsObject()) {
+    return ret;
+  }
+  struct to_copy_element {
+    string name;
+    Value const *v;
+  };
+  struct stack_element {
+    // Invariant: `it1` is always a iterator into the members of a json object,
+    // and that json object is part of the output document which is later
+    // returned from this function.
+    Value::ConstMemberIterator it1;
+    // Invariant: `v1` is always a json object and part of the output document
+    // which is later returned from this function.
+    Value *v1;
+    // Invariant: `v2` is always a json object and part of the 2nd input
+    // document to this function.
+    Value const *v2;
+    // Remember the elements to be copied from the 2nd input document into the
+    // document `ret` to be returned.
+    deque<to_copy_element> to_copy;
+  };
+  deque<stack_element> stack;
+  stack.push_back(stack_element{ret.MemberBegin(), &ret, &v2});
+
+  while (stack.size() > 0) {
+    auto &se = stack.back();
+    if (se.it1 != se.v1->MemberEnd()) {
+      // Naming scheme:
+      // m for 'member'
+      // (1|2) if it is out of 1st or 2nd input document.
+      // (n|v) name or value.
```

Comment:
:+1: The comments below are very helpful. 
Variable names should not require a key though. How about using `first_doc_member_name`, `second_doc_member_value` etc?